internal/services/managedapplications/managed_application_resource.go

```diff
@@ -204,97 +202,98 @@ func resourceManagedApplicationRead(d *pluginsdk.ResourceData, meta interface{})
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.ApplicationID(d.Id())
+	id, err := applications.ParseApplicationID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	resp, err := client.Get(ctx, id.ResourceGroup, id.Name)
+	resp, err := client.Get(ctx, *id)
 	if err != nil {
-		if utils.ResponseWasNotFound(resp.Response) {
+		if response.WasNotFound(resp.HttpResponse) {
 			log.Printf("[INFO] Managed Application %q does not exist - removing from state", d.Id())
 			d.SetId("")
 			return nil
 		}
-		return fmt.Errorf("failed to read Managed Application %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+		return fmt.Errorf("failed to read Managed Application %s: %+v", id, err)
 	}
 
-	d.Set("name", id.Name)
-	d.Set("resource_group_name", id.ResourceGroup)
-	d.Set("location", location.NormalizeNilable(resp.Location))
-	d.Set("kind", resp.Kind)
-	if err := d.Set("plan", flattenManagedApplicationPlan(resp.Plan)); err != nil {
-		return fmt.Errorf("setting `plan`: %+v", err)
-	}
-	if props := resp.ApplicationProperties; props != nil {
-		id, err := resourcesParse.ResourceGroupIDInsensitively(*props.ManagedResourceGroupID)
+	if m := resp.Model; m != nil {
+		p := m.Properties
+
+		d.Set("name", m.Name)
+		d.Set("resource_group_name", id.ResourceGroupName)
+		d.Set("location", location.NormalizeNilable(m.Location))
+		d.Set("kind", m.Kind)
+		if err := d.Set("plan", flattenManagedApplicationPlan(m.Plan)); err != nil {
+			return fmt.Errorf("setting `plan`: %+v", err)
+		}
+
+		id, err := resourcesParse.ResourceGroupIDInsensitively(*p.ManagedResourceGroupId)
 		if err != nil {
 			return err
 		}
 
 		d.Set("managed_resource_group_name", id.ResourceGroup)
-		d.Set("application_definition_id", props.ApplicationDefinitionID)
+		d.Set("application_definition_id", p.ApplicationDefinitionId)
 
-		parameterValues, err := flattenManagedApplicationParameterValuesValueToString(props.Parameters)
+		parameterValues, err := flattenManagedApplicationParameterValuesValueToString(p.Parameters)
 		if err != nil {
 			return fmt.Errorf("serializing JSON from `parameter_values`: %+v", err)
 		}
 		d.Set("parameter_values", parameterValues)
 
-		parameters, err := flattenManagedApplicationParametersOrOutputs(props.Parameters)
+		parameters, err := flattenManagedApplicationParametersOrOutputs(p.Parameters)
 		if err != nil {
 			return err
 		}
 		if err = d.Set("parameters", parameters); err != nil {
 			return err
 		}
 
-		outputs, err := flattenManagedApplicationParametersOrOutputs(props.Outputs)
+		outputs, err := flattenManagedApplicationParametersOrOutputs(p.Outputs)
 		if err != nil {
 			return err
 		}
 		if err = d.Set("outputs", outputs); err != nil {
 			return err
 		}
+
+		return tags.FlattenAndSet(d, m.Tags)
```

Comment:
(same here) pedantic, but having a single return point here removes bugs at PR review time:

```suggestion
		if err := tags.FlattenAndSet(d, m.Tags); err != nil {
			return fmt.Errorf("setting `tags`: %+v", err)
		}
```